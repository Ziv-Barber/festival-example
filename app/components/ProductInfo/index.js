// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import Countdown from '../../components/Countdown'
import MuiPaper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import ArrowDownward from '@material-ui/icons/ArrowDownward'
import styled from 'styled-components'
import withConditional from '../../utils/withConditional'
import type { ProductInfoPropsType } from './index.types'

/**
 * ProductInfo react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const ProductInfo = (props: ProductInfoPropsType) => {
  const { className } = props

  const ProductTextC = (props: {
    className: string,
    children?: React.Node
  }) => <span className={props.className}>{props.children}</span>
  const ProductText = withConditional(ProductTextC)
  const ConditionalCountdown = withConditional(Countdown)

  return (
    <MuiPaper
      elevation={0}
      square={true}
      className={className || 'productInfo'}
    >
      <ConditionalCountdown condition={props.date} date={props.date} />
      <ProductText condition={props.text} className="productText">
        {props.text}
      </ProductText>
      <Typography className="downArrow productDownArrow">
        {props.arrow || <ArrowDownward />}
      </Typography>
    </MuiPaper>
  )
}

const StyledProductInfo = styled(ProductInfo)`
  min-height: 40vh;
  background-size: cover;
  background-position: center bottom;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  .productText {
    text-align: center;
  }

  .downArrow {
    text-align: center;
  }

  .countdownRoot {
    display: flex;
    justify-content: center;
    background-color: transparent;
    box-shadow: none;
  }

  .countdownBox {
    width: 70px;
    margin: ${(props) => props.theme.spacing.unit / 2}px !important;
    padding: ${(props) => props.theme.spacing.unit}px;
    display: flex;
    flex-direction: column;
  }

  .countdownValue {
    text-align: center;
  }

  .countdownText {
    text-align: center;
  }
`

export default StyledProductInfo
