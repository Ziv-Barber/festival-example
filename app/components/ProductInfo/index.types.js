// @flow

export type ProductInfoPropsType = {
  className?: any,
  date?: Date,
  text?: string,
  arrow?: any,
  classes?: {
    root?: Object,
    countdownRoot?: Object,
    countdownBox?: Object,
    countdownValue?: Object,
    countdownText?: Object,
    productText?: Object,
    countdownRoot?: Object,
    arrow?: Object
  }
}
