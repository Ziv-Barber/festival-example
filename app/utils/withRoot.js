// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
