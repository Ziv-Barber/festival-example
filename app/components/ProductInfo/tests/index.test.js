// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import ProductInfo from '../index'
import withRoot from '../../../utils/withRoot'
import { IntlProvider } from 'react-intl'

// For flow:
let { describe, it, afterEach, expect } = global

const RealDate = global.Date

/** @test {Info} */
describe('<ProductInfo />', () => {
  afterEach(() => {
    global.Date = RealDate
    cleanup()
  })

  /** @test {Info} */
  it('should render', () => {
    // We need to mock the timer for the Countdown component:
    global.Date = class extends Date {
      constructor(reqDate) {
        super(reqDate)
        return reqDate ? this : new RealDate('2019-01-01')
      }
    }

    const TestComp = (prop) => (
      <IntlProvider locale="en">
        <ProductInfo text="This is a test" date={new RealDate('2020-05-01')} />
      </IntlProvider>
    )

    const App = withRoot(TestComp)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
