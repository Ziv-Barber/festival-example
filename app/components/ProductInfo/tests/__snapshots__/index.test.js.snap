// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductInfo /> should render 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 ProductInfo__StyledProductInfo-sc-16oieff-0 cgIMjA"
  >
    <div
      class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownRoot"
    >
      <div
        class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownBox"
      >
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownValue"
        >
          1
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownText"
        >
          <span>
            Year
          </span>
        </p>
      </div>
      <div
        class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownBox"
      >
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownValue"
        >
          121
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownText"
        >
          <span>
            Days
          </span>
        </p>
      </div>
      <div
        class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownBox"
      >
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownValue"
        >
          0
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownText"
        >
          <span>
            Hours
          </span>
        </p>
      </div>
      <div
        class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownBox"
      >
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownValue"
        >
          0
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownText"
        >
          <span>
            Min
          </span>
        </p>
      </div>
      <div
        class="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 countdownBox"
      >
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownValue"
        >
          0
        </p>
        <p
          class="MuiTypography-root-28 MuiTypography-body2-36 countdownText"
        >
          <span>
            Sec
          </span>
        </p>
      </div>
    </div>
    <span
      class="productText"
    >
      This is a test
    </span>
    <p
      class="MuiTypography-root-28 MuiTypography-body2-36 downArrow productDownArrow"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root-64"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
        />
      </svg>
    </p>
  </div>
</div>
`;
